shell/browser/web_view_guest_delegate.cc

```diff
@@ -76,15 +72,18 @@ content::WebContents* WebViewGuestDelegate::GetOwnerWebContents() {
 void WebViewGuestDelegate::OnZoomChanged(
     const WebContentsZoomController::ZoomChangedEventData& data) {
   if (data.web_contents == GetOwnerWebContents()) {
+    auto* zoom_controller = api_web_contents_->GetZoomController();
     if (data.temporary) {
-      api_web_contents_->GetZoomController()->SetTemporaryZoomLevel(
-          data.new_zoom_level);
+      zoom_controller->SetTemporaryZoomLevel(data.new_zoom_level);
     } else {
-      api_web_contents_->GetZoomController()->SetZoomLevel(data.new_zoom_level);
+      if (blink::PageZoomValuesEqual(data.new_zoom_level,
+                                     zoom_controller->GetZoomLevel()))
```

Comment:
I don't fully understand temporary zoom so this suggestion may be wrong, but...

wouldn't we want this is-the-value-unchanged check to go before `if (data.temporary) {` so that it's always called regardless of the value of `data.temporary`?

Everything else LGTM. And maybe this is good too as-is; just asking because temporary zoom looks complicated (e.g. the use of temporary zoom in `WebContentsZoomController::SetZoomMode()` :smile_cat:)

Reply:
@ckerr i thought about this but decided against it since a webContents can have an existing temporary zoom at the time it's set, which may or may not differ from the current non-temporary zoom. In that case, we could get a scenario where, let's say, zoom is 1.2 and temporary zoom is 1.5. We call `WebViewGuestDelegate::OnZoomChanged` with a new value of `1.2` - with what you're suggesting above, we would exit early in this case, but actually we _should_ be changing the temporary zoom from 1.5 to 1.2.